forte/data/data_store.py

```diff
@@ -1315,6 +1329,111 @@ def _get_existing_ann_entry_tid(self, entry: List[Any]):
                 "getting entry id for annotation-like entry."
             )
 
+    def get_attribute_positions(self, type_name: str) -> Dict[str, int]:
+        r"""This function returns a dictionary where the key represents
+        the attributes of the entry of type ``type_name`` and value
+        represents the index of the position where this attribute is
+        stored in the data store entry of this type.
+        For example:
+
+        .. code-block:: python
+
+            positions = data_store.get_attribute_positions(
+                    "ft.onto.base_ontology.Document"
+                )
+
+            # positions = {
+            #   "begin": 2,
+            #    "end": 3,
+            #    "payload_idx": 4,
+            #    "document_class": 5,
+            #    "sentiment": 6,
+            #    "classifications": 7
+            # }
+
+        Args:
+            type_name (str): The fully qualified type name of a type.
+
+        Returns:
+            A dictionary indicating the attributes of an entry of type
+            ``type_name`` and their respective positions in a data store
+            entry.
+        """
+        type_data = self._get_type_info(type_name)
+
+        positions: Dict[str, int] = {}
+        for attr, val in type_data[constants.ATTR_INFO_KEY].items():
+            positions[attr] = val[constants.ATTR_INDEX_KEY]
+
+        return positions
+
+    def transform_data_store_entry(self, entry: List[Any]) -> Dict:
+        r"""
+        This method converts a raw data store entry into a format more easily
+        understandable to users. Data Store entries are stored as lists and
+        are not very easily understandable. This method converts ``DataStore``
+        entries from a list format to a dictionary based format where the key
+        is the names of the attributes of an entry and the value is the values
+        corresponding attributes in the data store entry.
+        For example:
+
+        .. code-block:: python
+
+            # Entry of type 'ft.onto.base_ontology.Sentence'
```

Comment:
The more conventional way to write this docstring is the `doctest` format (https://docs.python.org/3/library/doctest.html). We actually are testing the correctness of these automatically here: https://github.com/asyml/forte/blob/master/.github/workflows/main.yml#L99